---
title: "**RProject**: Transcriptional and proteomic insights into the host response in fatal COVID-19 cases"
subtitle: "<span style='color: #EB8B89;'>Bioinformatics I</span>"
author: "Ilaria Coratella"
output:
  prettydoc::html_pretty:
    theme: leonids
---

```{r setup, include=FALSE}
library(DESeq2)
library(pheatmap)
library(RColorBrewer)
library(EnhancedVolcano)
library(ggplot2)
library(AnnotationDbi)
library(org.Hs.eg.db)
library(clusterProfiler)
suppressMessages(suppressWarnings(library(clusterProfiler)))
knitr::opts_chunk$set(echo = FALSE)
setwd("C:/Users/Ilaria/Desktop/UNI/not_passed/BI1/ProjectR")
```

## Introduction

![](immagine.png)

<p style="line-height: 1.2;">
I chose this article, which analyzes pathways related to neutrophil activation and pulmonary fibrosis among the major up-regulated transcriptional signatures in lung tissue obtained from patients who died of COVID-19 in Wuhan, China. Examination of the colonic transcriptome of these patients suggested that SARS-CoV-2 impacted host responses even at a site with no obvious pathogenesis.
</p>



## Covid 19

<p style="line-height: 1.2;">
SARS-CoV-2, the etiologic agent for coronavirus disease 2019 (COVID-19), emerged in Wuhan, China, in late December 2019. Since then, more than 29.5 million people have been
infected worldwide, with over 933,000 deaths as of September 15, 2020. Although research into the clinical and scientific features of SARS-CoV-2 pathogenesis has progressed at a dizzying pace, pathogenesis of COVID-19 is still incompletely understood.
</p>

## What is the aim of this project?

<p style="line-height: 1.2;">
The aim of this presentation is to show the results of two differential expression analysis, hence to find DEGs in lung and colon tissues of diseased patients. I preferred to perform two symultaneous analysis for the two tissues in order to obtain a more precise result.
</p>

## Filtering of the original dataset

<p style="line-height: 1.2;">
Firstly, I have filtered my original experimental design dataset for tissue, and I did it both for lung and colon.
</p>

```{r, echo = TRUE}
setwd("C:/Users/Ilaria/Desktop/UNI/not_passed/BI1/ProjectR")
library(readxl)
library(writexl)

df <- read_excel("E-ENAD-46-experiment-design - Copiaexcel.xlsx")
df_lung <- df[df$`Sample Characteristic[organism part]` == 'lung', ]
write_xlsx(df_lung, "new_experiment_design_lung_R.xlsx")

df_colon <- df[df$`Sample Characteristic[organism part]` == 'colon', ]
write_xlsx(df_colon, "new_experiment_design_colon_R.xlsx")
```

## Filtered dataset for lung
![](lung_excel.png)


## Filtered dataset for colon
![](colon_excel.png)


## Filtering of the genes

<p style="line-height: 1.2;">
Once I have separated the samples on the basis of the tissue type, I divided the counts of the genes. This filtering operation has been carried out on the basis of the names of the samples previously divided:
</p>
```{r, echo = TRUE}
lung_design <- readxl::read_excel("new_experiment_design_lung_R.xlsx")
colon_design <- readxl::read_excel("new_experiment_design_colon_R.xlsx")

lung_sample <- lung_design$Run
colon_sample <- colon_design$Run


count <- readxl::read_excel("E-ENAD-46-raw-counts.xlsx")

lung_counts <- count[,c("Gene ID","Gene Name",lung_sample)]
colon_counts <- count[,c("Gene ID","Gene Name",colon_sample)]

write.table(lung_counts, file = "lung_counts.tsv", quote = F, row.names = F, sep = "\t")
write.table(colon_counts,file = "colon_counts.tsv", quote = F, row.names = F, sep = "\t")

write.table(lung_design, file = "design_lung.tsv", quote = F, row.names = F, sep = "\t")
write.table(colon_design, file = "design_colon.tsv", quote = F, row.names = F, sep = "\t")
```

# <span style="color: #D27D7B;">Differential expression analysis </span>:
## Importing data as matrices
### I obtained a total of 19 samples, 9 diseased and 10 healthy.

<p style="line-height: 1.2;">
First of all I have uploaded the data of the study, both the study design and the gene counts. I did it both for <span style="color: #D27D7B;">LUNG</span> and <span style="color: #7AB1D1;">COLON</span>:

<span style="color: #D27D7B;">**LUNG**</span>
</p>

```{r, echo = TRUE}
coldata <- read.delim('design_lung.tsv', header = TRUE, row.names = 1) 
#export as a matrix: it creates a matrix with the data of the experiment design file.

rowcounts <- read.delim('lung_counts.tsv', header = TRUE, row.names = 1)
#export as a matrix: it creates a matrix with the data of the raw counts file.

```
<p style="line-height: 1.2;">

<span style="color: #7AB1D1;">**COLON**</span>
</p>

```{r, echo = TRUE}
coldata1 <- read.delim('design_colon.tsv', header = TRUE, row.names = 1) 
#export as a matrix: it creates a matrix with the data of the experiment design file.

rowcounts1 <- read.delim('colon_counts.tsv', header = TRUE, row.names = 1)
#export as a matrix: it creates a matrix with the data of the raw counts file.

```


## Removing the 'extra' column

<p style="line-height: 1.2;">
Since I noticed that there were two columns to identify the genes, i removed one. In fact, just one column is necessary to perform the Differential Expression Analysis. This is the code to remove the column <span style="color: #D27D7B;">Gene.Name</span>:

<span style="color: #D27D7B;">**LUNG**</span>
</p>
```{r, echo = TRUE}

countData <- as.matrix(subset(rowcounts, select = c(-Gene.Name)))

```

<p style="line-height: 1.2;">

<span style="color: #7AB1D1;">**COLON**</span>
</p>

```{r, echo = TRUE}

countData1 <- as.matrix(subset(rowcounts1, select = c(-Gene.Name)))

```

## Specify the Sample Type

<p style="line-height: 1.2;">
This function is used to specify to the program in which column of the <span style="color: #D27D7B;">coldata</span> (experiment design file) it is indicated whether the samples are disease or control.

<span style="color: #D27D7B;">**LUNG**</span>
</p>
```{r, echo = TRUE}

designFormula <- "~ Sample.Characteristic.disease."

```

<p style="line-height: 1.2;">

<span style="color: #7AB1D1;">**COLON**</span>
</p>
```{r, echo = TRUE}

designFormula <- "~ Sample.Characteristic.disease."

```



## RNAseq analysis

<p style="line-height: 1.2;">
this lines of code can create a DESeq dataset object from the count matrix and the colData, which I named <span style="color: #D27D7B;">dds</span>. As it can easily be observed from the output, <span style="color: #D27D7B;">I have 58375 genes</span>. <span style="color: #D27D7B;">As.formula()</span> can be used to convert the string into a formula object, which can be useful in order to perform statistical analysis.

<span style="color: #D27D7B;">**LUNG**</span>
</p>
```{r, echo = TRUE, warning=FALSE, message=FALSE}
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = coldata,
                              design = as.formula(designFormula))
print(dds)
```

<p style="line-height: 1.2;">

<span style="color: #7AB1D1;">**COLON**</span>
</p>

```{r, echo = TRUE, warning=FALSE, message=FALSE}
dds1 <- DESeqDataSetFromMatrix(countData = countData1,
                              colData = coldata1,
                              design = as.formula(designFormula))
print(dds1)
```


## RNAseq analysis

<p style="line-height: 1.2;">
In order to obtain a more precise result in the end, I decide to filter the genes, keeping only the ones which are expressed more than once. This function saves the genes with 0 expression in all the samples as <span style="color: #D27D7B;">FALSE</span>, and all the others as <span style="color: #D27D7B;">TRUE</span>:

<span style="color: #D27D7B;">**LUNG**</span>
</p>
```{r, echo = TRUE}
expressed <-rowSums(counts(dds)) > 1
table(expressed)
```

<p style="line-height: 1.2;">

<span style="color: #7AB1D1;">**COLON**</span>

</p>
```{r, echo = TRUE}
expressed1 <-rowSums(counts(dds1)) > 1
table(expressed1)
```


## RNAseq analysis

<p style="line-height: 1.2;">
Then, I removed my FALSE values from my dds, so that now it will contain only the genes that i have selected.To be more precise.

<span style="color: #D27D7B;">**LUNG**</span>
If previously I had 58375 genes, <span style="color: #D27D7B;">now I have 41449 genes</span>:
</p>

```{r, echo = TRUE}
dds <- dds[expressed, ]
print(dds)
```

<p style="line-height: 1.2;">
<span style="color: #7AB1D1;">**COLON**</span>
if previously I had 58375 genes, <span style="color: #D27D7B;">now I have 42367 genes</span>:
</p>
```{r, echo = TRUE}
dds1 <- dds1[expressed1, ]
print(dds1)
```


## RNAseq analysis
<p style="line-height: 1.2;">
Afterward, I continued with the pipeline by applying <span style="color: #D27D7B;">DESeq()</span>, which performs count normalization and estimates dispersion values:

<span style="color: #D27D7B;">**LUNG**</span>
</p>
```{r, echo = TRUE, warning=FALSE, message=FALSE}
dds <- DESeq(dds)
```

<p style="line-height: 1.2;">

<span style="color: #7AB1D1;">**COLON**</span>
</p>
```{r, echo = TRUE, warning=FALSE, message=FALSE}
dds1 <- DESeq(dds1)
```


<p style="line-height: 1.2;">
It carries out the analysis, and then overwrites the results in <span style="color: #D27D7B;">dds</span>.
</p>


## RNAseq analysis
<p style="line-height: 1.2;">
Then, I proceeded to create a PCA plot by utilizing the <span style="color: #D27D7B;">vst()</span> function for normalization. I saved it into another object, called <span style="color: #D27D7B;">rld</span>.

<span style="color: #D27D7B;">**LUNG**</span>
</p>
```{r, echo = TRUE, warning=FALSE, message=FALSE}
rld <- vst(dds)
```

<p style="line-height: 1.2;">

<span style="color: #7AB1D1;">**COLON**</span>
</p>
```{r, echo = TRUE, warning=FALSE, message=FALSE}
rld1 <- vst(dds1)
```

## RNAseq analysis
<p style="line-height: 1.2;">
Then i can obtain a PCA plot the normalized values I obtain thanks to the previous lines of code

<span style="color: #D27D7B;">**LUNG**</span>
</p>
```{r, echo = TRUE, warning=FALSE, message=FALSE}

DESeq2::plotPCA(rld, intgroup = 'Sample.Characteristic.disease.')

```

<p style="line-height: 1.2;">
As it is possible to observe, almost all of the samples seem to be correctly classified, with the exception of one sample classified as COVID-19 patient. It may be the result of an error, since it shows a considerable high variance with respect to the other samples.
</p>

<p style="line-height: 1.2;">

<span style="color: #7AB1D1;">**COLON**</span>
</p>

```{r, echo = TRUE, warning=FALSE, message=FALSE}

DESeq2::plotPCA(rld1, intgroup = 'Sample.Characteristic.disease.')

```

<p style="line-height: 1.2;">
This PCA plot is messier than the previous one. We can observe a certain kind of separation between the two categories of genes, but the variance is high and COVID-19 and normal samples are close the one with the other. It is probable that, affecting COVID-19 mainly lungs, a difference in gene expression can be easily detect in lungs, but less easily in colon. However, it is interesting that a slight difference in gene expression does still exists, and this is one of the main points of the paper.
</p>



## RNAseq analysis
<p style="line-height: 1.2;">
to create an object which describes how different are couples of samples. It is important to build the <span style="color: #D27D7B;">heatmap</span> of sample sample distance:

<span style="color: #D27D7B;">**LUNG**</span>
</p>
```{r, echo = TRUE, warning=FALSE, message=FALSE}
SampleDists <- dist(t(assay(rld)))
SampleDistMatrix <- as.matrix(SampleDists)
```

![](sampledistmatrix_lung.png)
<p style="line-height: 1.2;">

<span style="color: #7AB1D1;">**COLON**</span>
</p>
```{r, echo = TRUE, warning=FALSE, message=FALSE}
SampleDists1 <- dist(t(assay(rld1)))
SampleDistMatrix1 <- as.matrix(SampleDists1)
```

![](sampledistmatrix_colon.png)
## RNAseq analysis
Now I can plot my heatmap:

<span style="color: #D27D7B;">**LUNG**</span>

```{r, echo = TRUE, warning=FALSE, message=FALSE}
library(viridis)  
pheatmap(SampleDistMatrix,
         main = "Heatmap of sample-sample distance",
         clustering_distance_rows = SampleDists,
         clustering_distance_cols = SampleDists,
         col = magma(9))
```

<p style="line-height: 1.2;">
However, we can easily observe a sample, SRR12816735, which is dissimilar to every other sample, and it may be the same we observed before in the PCA plot. Since it is highly possible that it is <span style="color: #D27D7B;">misclassified</span>, I prefer proceeding removing it.
</p>

```{r, echo = TRUE, warning=FALSE, message=FALSE}
coldata <- coldata[-18, ]
rowcounts <- rowcounts[, -19]
```


## RNAseq analysis
<p style="line-height: 1.2;">
Now I obtained the same graphs, but without SRR12816735:
</p>
```{r, echo = FALSE, warning=FALSE, message=FALSE}
countData <- as.matrix(subset(rowcounts, select = c(-Gene.Name)))
designFormula <- "~ Sample.Characteristic.disease."

#create a DESeq dataset object from the count matrix and the colData
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = coldata,
                              design = as.formula(designFormula))
#filter Genes
#filter Low Expressed genes with "RowSums(Counts(dds)) > 1" si salva i geni che hanno zero espressione in tutti i sample come FALSE, vice versa come TRUE
expressed <-rowSums(counts(dds)) > 1
dds <- dds[expressed, ] 

#display results #13358 fALSE #45377 TRUE
#table(expressed)

#DESeq Pipeline
#run DESeq2 pipeline #IL COMANDO Deseq ti fa l'analisi. ho sovrascritto i risultati in dds.
dds <- DESeq(dds)

#PCA of the normalized values using the vst() function. normalizzazione.e l'ho salvata in un altro oggetto
rld <- vst(dds)
DESeq2::plotPCA(rld, intgroup = 'Sample.Characteristic.disease.')

#Heatmap ----
#plot an Heatmap of the sample-to-sample distances
#sample sample distance
SampleDists <- dist(t(assay(rld)))
#SampleDists

# to create an object which describes how different are couples of samples.
SampleDistMatrix <- as.matrix(SampleDists)
```


```{r, echo = FALSE, warning=FALSE, message=FALSE}
library(viridis)  
pheatmap(SampleDistMatrix,
         main = "LUNG Heatmap of sample-sample distance",
         clustering_distance_rows = SampleDists,
         clustering_distance_cols = SampleDists,
         col = magma(9))
```
<p style="line-height: 1.2;">
We can observe a more homogeneous PCA and heatmap.
</p>
<p style="line-height: 1.2;">

<span style="color: #7AB1D1;">**COLON**</span>
</p>
```{r, echo = TRUE, warning=FALSE, message=FALSE}
library(viridis)  
pheatmap(SampleDistMatrix1,
         main = "COLON Heatmap of sample-sample distance",
         clustering_distance_rows = SampleDists1,
         clustering_distance_cols = SampleDists1,
         col = magma(9))
```

<p style="line-height: 1.2;">
The heatmap of the colon does not show very clean results.
</p>
## RNAseq analysis
<p style="line-height: 1.2;">
Than I performed the comparison between the two groups. I specified the coloumn of sampledata (experimental design) that has the group classification: COVID-19 vs normal. 

<span style="color: #D27D7B;">**LUNG**</span>
3447 genes are over overexpressed in dds while 4556 underexpressed.
</p>

```{r, echo = TRUE, warning=FALSE, message=FALSE}
res <- results(dds, contrast = c("Sample.Characteristic.disease.", 'COVID-19', 'normal'))
summary(res)
```
<p style="line-height: 1.2;">

<span style="color: #7AB1D1;">**COLON**</span>
3447 genes are over overexpressed in DS while 4556 underexpressed (cambia!).
</p>
```{r, echo = TRUE, warning=FALSE, message=FALSE}
res1 <- results(dds1, contrast = c("Sample.Characteristic.disease.", 'COVID-19', 'normal'))
summary(res1)
```


## VOLCANO PLOT

<p style="line-height: 1.2;">
This is my volcano plot:

<span style="color: #D27D7B;">**LUNG**</span>
</p>
```{r, echo = TRUE, warning=FALSE, message=FALSE}
EnhancedVolcano(
  as.data.frame(res),
  labSize = 2.5,
  title = "VolcanoPlot",
  subtitle = "COVID-19 vs normal",
  lab = rownames(res),
  FCcutoff = 2,
  x = 'log2FoldChange',
  y = 'pvalue',
  col = c("#b4ddd4", "#f8a0b1", "#088490", "#f92e5d")  
)

```

<p style="line-height: 1.2;">

<span style="color: #7AB1D1;">**COLON**</span>
</p>
```{r, echo = FALSE, warning=FALSE, message=FALSE}
EnhancedVolcano(
  as.data.frame(res1),
  labSize = 2.5,
  title = "VolcanoPlot",
  subtitle = "COVID-19 vs normal",
  lab = rownames(res1),
  FCcutoff = 2,
  x = 'log2FoldChange',
  y = 'pvalue',
  col = c("#b4ddd4", "#f8a0b1", "#088490", "#f92e5d")  
)

```

## MAPLOT

<p style="line-height: 1.2;">
This is my MaPlot plot:

<span style="color: #D27D7B;">**LUNG**</span>
</p>
```{r, echo = TRUE, warning=FALSE, message=FALSE}
DESeq2::plotMA(object = res, 
               main ='MAPlot Result',
               colSig = "#f8a0b1",
               colNonSig = "#f92e5d",
               colLine = "#b4ddd4",
               xlab="mean of normalized counts", 
               ylab="log2FoldChange",
               ylim = c(-12, 12))

```

<p style="line-height: 1.2;">

<span style="color: #7AB1D1;">**COLON**</span>
</p>
```{r, echo = FALSE, warning=FALSE, message=FALSE}
DESeq2::plotMA(object = res1, 
               main ='MAPlot Result',
               colSig = "#f8a0b1",
               colNonSig = "#f92e5d",
               colLine = "#b4ddd4",
               xlab="mean of normalized counts", 
               ylab="log2FoldChange",
               ylim = c(-12, 12))

```

## Comparison with Expression Atlas results: DOWN-regulated <span style="color: #D27D7B;">**LUNG**</span> genes


<p style="line-height: 1.2;">

C2orf91

As it is observable, we can see a log2foldchange of -8.27, while in the result of the experiment in Expression Atlas it is -7.1

</p>
```{r, echo = TRUE, warning=FALSE, message=FALSE}
res['ENSG00000205086',]
```

<p style="line-height: 1.2;">

UPK3B

As it is observable, we can see a log2foldchange of -8.25, while in the result of the experiment in Expression Atlas it is -6.8
</p>
```{r, echo = TRUE, warning=FALSE, message=FALSE}
res['ENSG00000243566',]
```

## Comparison with Expression Atlas results: UP-regulated <span style="color: #D27D7B;">**LUNG**</span> genes

<p style="line-height: 1.2;">
SERPINB4
As it is observable, we can see a log2foldchange of 8.34, while in the result of the experiment in Expression Atlas it is 6.4:
</p>
```{r, echo = TRUE, warning=FALSE, message=FALSE}
res['ENSG00000206073',]
```
<p style="line-height: 1.2;">
CYP19A1
As it is observable, we can see a log2foldchange of 6.96, while in the result of the experiment in Expression Atlas it is 5.7:
</p>
```{r, echo = TRUE, warning=FALSE, message=FALSE}
res['ENSG00000137869',]
```


# Some questions on DEGs
## How many genes are differentially expressed at the thresholds of padj < 0.1?

<span style="color: #D27D7B;">**LUNG**</span>
```{r, echo = TRUE, warning=FALSE, message=FALSE}
new_res <- res[!is.na(res$padj),]
summary(new_res[new_res$padj < 0.1 & abs(new_res$log2FoldChange) > 0 ,])
```
<p style="line-height: 1.2;">
We can observe 8003 Differentially Expressed Genes with an adjusted p-value of 0.1.
</p>
<p style="line-height: 1.2;">
<span style="color: #7AB1D1;">**COLON**</span>
</p>
```{r, echo = TRUE, warning=FALSE, message=FALSE}
new_res1 <- res1[!is.na(res1$padj),]
summary(new_res1[new_res1$padj < 0.1 & abs(new_res1$log2FoldChange) > 0 ,])
```
<p style="line-height: 1.2;">
We can observe 10330 Differentially Expressed Genes with an adjusted p-value of 0.1.
</p>

## How many genes are differentially expressed at the thresholds of padj < 0.05?

<span style="color: #D27D7B;">**LUNG**</span>
```{r, echo = TRUE, warning=FALSE, message=FALSE}
summary(new_res[new_res$padj < 0.05 & abs(new_res$log2FoldChange) > 0 ,])
```
<p style="line-height: 1.2;">
We can observe 6037 Differentially Expressed Genes with an adjusted p-value of 0.05.
</p>
<p style="line-height: 1.2;">
<span style="color: #7AB1D1;">**COLON**</span>
</p>
```{r, echo = TRUE, warning=FALSE, message=FALSE}
summary(new_res1[new_res1$padj < 0.05 & abs(new_res1$log2FoldChange) > 0 ,])
```
<p style="line-height: 1.2;">
We can observe 7934 Differentially Expressed Genes with an adjusted p-value of 0.05.
</p>

## How many genes at the thresholds of padj < 0.05 are UP-regulated ( > 1)?

<span style="color: #D27D7B;">**LUNG**</span>
```{r, echo = TRUE, warning=FALSE, message=FALSE}
summary(new_res[new_res$padj < 0.05 & new_res$log2FoldChange > 1  ,])
```
<p style="line-height: 1.2;">
At a padj of 0.05 and using as log2FoldChange threshold the value 1, we can observe 1539 UP-regulated genes.
</p>
<p style="line-height: 1.2;">
<span style="color: #7AB1D1;">**COLON**</span>
</p>
```{r, echo = TRUE, warning=FALSE, message=FALSE}
summary(new_res1[new_res1$padj < 0.05 & new_res1$log2FoldChange > 1  ,])
```
<p style="line-height: 1.2;">
At a padj of 0.05 and using as log2FoldChange threshold the value 1, we can observe 1232 UP-regulated genes.
</p>

## How many genes at the thresholds of padj < 0.05 are DOWN-regulated ( < -1)?

<span style="color: #D27D7B;">**LUNG**</span>
```{r, echo = TRUE, warning=FALSE, message=FALSE}
summary(new_res[new_res$padj < 0.05 & new_res$log2FoldChange < -1  ,])
```
<p style="line-height: 1.2;">
At a padj of 0.05 and using as log2FoldChange threshold the value -1, we can observe 2930 DOWN-regulated genes.
</p>
<p style="line-height: 1.2;">
<span style="color: #7AB1D1;">**COLON**</span>
</p>
```{r, echo = TRUE, warning=FALSE, message=FALSE}
summary(new_res1[new_res1$padj < 0.05 & new_res1$log2FoldChange < -1  ,])
```
<p style="line-height: 1.2;">
At a padj of 0.05 and using as log2FoldChange threshold the value -1, we can observe 5551 DOWN-regulated genes.
</p>

# Gene Ontology (GO) Enrichment analysis
## UP-REGULATED

<span style="color: #D27D7B;">**LUNG**</span>
<p style="line-height: 1.2;">
Firstly, I have to find the column in which the gene symbol is saved:
</p>

```{r, echo = TRUE, warning=FALSE, message=FALSE}
resSig <- subset(new_res, padj < 0.1)
upReg <- subset(resSig, log2FoldChange > 1)
upReg$symbol <- mapIds(org.Hs.eg.db,
                       keys=rownames(upReg),
                       column="SYMBOL",
                       keytype="ENSEMBL",
                       multiVals="first")
```

```{r, echo = FALSE, warning=FALSE, message=FALSE}
resSig1 <- subset(new_res1, padj < 0.1)
upReg1 <- subset(resSig1, log2FoldChange > 1)
upReg1$symbol <- mapIds(org.Hs.eg.db,
                       keys=rownames(upReg1),
                       column="SYMBOL",
                       keytype="ENSEMBL",
                       multiVals="first")
```

## Gene Ontology (GO) Enrichment analysis

<p style="line-height: 1.2;">
Secondly, we perform the Enrichment analysis thanks to the function enrichGO(). I decided to perform it for Biological Process (BP) and Molecular Function (MF):
</p>

```{r, echo = TRUE, warning=FALSE, message=FALSE}
GO_BP_UP <- enrichGO(upReg$symbol, OrgDb = "org.Hs.eg.db", 
                     keyType = "SYMBOL", ont = "BP")
```

```{r, echo = TRUE, warning=FALSE, message=FALSE}
GO_MF_UP <- enrichGO(upReg$symbol, OrgDb = "org.Hs.eg.db", 
                       keyType = "SYMBOL", ont = "MF")
```

```{r, echo = FALSE, warning=FALSE, message=FALSE}
GO_BP_UP1 <- enrichGO(upReg1$symbol, OrgDb = "org.Hs.eg.db", 
                     keyType = "SYMBOL", ont = "BP")
```

```{r, echo = FALSE, warning=FALSE, message=FALSE}
GO_MF_UP1 <- enrichGO(upReg1$symbol, OrgDb = "org.Hs.eg.db", 
                       keyType = "SYMBOL", ont = "MF")
```

### DOT PLOT FOR UP-REG Biological Process (BP) LUNG

```{r, echo = TRUE, warning=FALSE, message=FALSE}
dotplot(GO_BP_UP, title = "UpBioProcess")
```

### DOT PLOT FOR UP-REG Molecular Function (MF) LUNG

```{r, echo = TRUE, warning=FALSE, message=FALSE}
dotplot(GO_MF_UP, title = "UpMolFunctions")
```

### DOT PLOT FOR UP-REG Biological Process (BP) COLON

```{r, echo = TRUE, warning=FALSE, message=FALSE}
dotplot(GO_BP_UP1, title = "UpBioProcess")
```

### DOT PLOT FOR UP-REG Molecular Function (MF) COLON

```{r, echo = TRUE, warning=FALSE, message=FALSE}
dotplot(GO_MF_UP1, title = "UpMolFunctions")
```

# Gene Ontology (GO) Enrichment analysis

## DOWN-REGULATED

<p style="line-height: 1.2;">
<span style="color: #D27D7B;">**LUNG**</span>
I perform the same passages for down-regulated genes:
</p>


```{r, echo = TRUE, warning=FALSE, message=FALSE}
downReg <- subset(resSig, log2FoldChange < -1)
downReg$symbol <- mapIds(org.Hs.eg.db,
                         keys=rownames(downReg),
                         column="SYMBOL",
                         keytype="ENSEMBL",
                         multiVals="first")

GO_BP_DOWN <- enrichGO(downReg$symbol, OrgDb = "org.Hs.eg.db", 
                       keyType = "SYMBOL", ont = "BP")
GO_MF_DOWN <- enrichGO(downReg$symbol, OrgDb = "org.Hs.eg.db", 
                       keyType = "SYMBOL", ont = "MF")
dotplot(GO_BP_DOWN, title = "Downregualted Biological Process")
dotplot(GO_MF_DOWN, title = "Downregualted Molecular Functions")
```
<p style="line-height: 1.2;">

<span style="color: #7AB1D1;">**COLON**</span>
</p>
```{r, echo = TRUE, warning=FALSE, message=FALSE}
downReg1 <- subset(resSig1, log2FoldChange < -1)
downReg1$symbol <- mapIds(org.Hs.eg.db,
                         keys=rownames(downReg1),
                         column="SYMBOL",
                         keytype="ENSEMBL",
                         multiVals="first")

GO_BP_DOWN1 <- enrichGO(downReg1$symbol, OrgDb = "org.Hs.eg.db", 
                       keyType = "SYMBOL", ont = "BP")
GO_MF_DOWN1 <- enrichGO(downReg1$symbol, OrgDb = "org.Hs.eg.db", 
                       keyType = "SYMBOL", ont = "MF")
dotplot(GO_BP_DOWN1, title = "Downregualted Biological Process")
dotplot(GO_MF_DOWN1, title = "Downregualted Molecular Functions")
```

